graphics/src/gradient.rs

```diff
@@ -96,26 +96,33 @@ impl Linear {
     }
 
     /// Packs the [`Gradient`] for use in shader code.
-    pub fn pack(&self) -> [f32; 44] {
-        let mut pack: [f32; 44] = [0.0; 44];
+    pub fn pack(&self) -> Packed {
+        let mut data: [f32; 44] = [0.0; 44];
 
         for (index, stop) in self.stops.iter().enumerate() {
             let [r, g, b, a] =
                 stop.map_or(Color::default(), |s| s.color).into_linear();
 
-            pack[index * 4] = r;
-            pack[(index * 4) + 1] = g;
-            pack[(index * 4) + 2] = b;
-            pack[(index * 4) + 3] = a;
+            data[index * 4] = r;
+            data[(index * 4) + 1] = g;
+            data[(index * 4) + 2] = b;
+            data[(index * 4) + 3] = a;
 
-            pack[32 + index] = stop.map_or(2.0, |s| s.offset);
+            data[32 + index] = stop.map_or(2.0, |s| s.offset);
         }
 
-        pack[40] = self.start.x;
-        pack[41] = self.start.y;
-        pack[42] = self.end.x;
-        pack[43] = self.end.y;
+        data[40] = self.start.x;
+        data[41] = self.start.y;
+        data[42] = self.end.x;
+        data[43] = self.end.y;
 
-        pack
+        Packed { data }
     }
 }
+
+/// Packed [`Gradient`] data for use in shader code.
+#[derive(Debug)]
+pub struct Packed {
+    /// The packed [`Gradient`] data.
+    pub data: [f32; 44],
```

Comment:
Ideally, we should make this struct `repr(C)` and the `data` completely private.

Reply:
Good idea! Changed in e843b4f045a5a825e991346b1aaffe8eece3ffbe. The imports will be moved to more appropriate places in #1873